src/shared/bus-unit-util.c

```diff
@@ -464,6 +464,29 @@ static int bus_append_cgroup_property(sd_bus_message *m, const char *field, cons
                 return 1;
         }
 
+        if (streq(field, "CPUQuotaPeriod")) {
+                usec_t u = 0;
+
+                if (!isempty(eq)) {
+                        r = parse_time(eq, &u, MSEC_PER_SEC);
+                        if (r < 0) {
+                                log_error_errno(r, "CPU quota period '%s' invalid.", eq);
+                                return -EINVAL;
+                        }
+
+                        /* Valid range for period in cpu.max is [1ms, 1s]:
+                         * https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/kernel/sched/core.c?h=v4.18-rc5#n6545 */
+                        if (u < 1 * USEC_PER_MSEC || u > 1 * USEC_PER_SEC)
+                                return log_error_errno(-ERANGE, "CPU quota period '%s' outside of the valid range [1ms, 1s].", eq);
```

Comment:
so far we had the rule, that our logging calls would only log error codes that have been received from somewhere else, rather than error codes that are generated here. (@keszybz disagrees with that rule however, and maybe we should revisit this) However for now I think we should stick to this rule, hence please rework this bit to:

```c
if (…) {
        log_error("CPU quota…
        return -ERANGE;
}
```